### PR TITLE
[Lake] Add GroupAssigner for writing tests easier

### PR DESCRIPTION
**Pull request by sduzh**
Make TabletManager decoupled from Starlet for writing
tests easier

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: be/src/storage/lake/tablet_manager.h [hunk 1]
```diff
@@ -52,8 +48,29 @@ class TabletManager {
     StatusOr<MetadataIterator> list_txn_log(const std::string& group);
     StatusOr<MetadataIterator> list_txn_log(const std::string& group, int64_t tablet_id);
 
-    Starlet* _starlet;
+    GroupAssigner* _group_assigner;
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.h:51]: Starlet is also responsible for writing and reading objects.

File: be/src/storage/lake/tablet_manager.h [hunk 2]
```diff
@@ -52,8 +48,29 @@ class TabletManager {
...
+    //
+    // All implementations must be thread-safe.
+    virtual StatusOr<std::string> get_group(int64_t tablet_id) = 0;
+
+    // Return a list of storage groups.
+    // This may not be a full list of storage groups and may differed
+    // in each return.
+    //
+    // All implementations must be thread-safe.
+    virtual Status list_group(std::vector<std::string>* groups) = 0;
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.h:73]: When will it be used?

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.h:73]: GC

**Comment by sduzh** [comment]
run starrocks_be_unittest

File: be/src/storage/lake/tablet_manager.h [hunk 3]
```diff
@@ -52,8 +48,29 @@ class TabletManager {
     StatusOr<MetadataIterator> list_txn_log(const std::string& group);
     StatusOr<MetadataIterator> list_txn_log(const std::string& group, int64_t tablet_id);
 
-    Starlet* _starlet;
+    GroupAssigner* _group_assigner;
     std::unique_ptr<Cache> _metacache;
 };
 
+// GroupAssigner is used to assign a storage directory for a tablet.
+class GroupAssigner {
```
Comment (imay) [inline be/src/storage/lake/tablet_manager.h:56]: I'd like to make it in a single file

File: be/src/storage/lake/group_assigner.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+#include <string>
+#include <vector>
+
+#include "common/statusor.h"
+
+namespace starrocks::lake {
+
+// GroupAssigner is used to assign a storage directory for a tablet.
+class GroupAssigner {
```
Comment (imay) [inline be/src/storage/lake/group_assigner.h:13]: virutal ~GroupAssigner()?

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**